ldappool/__init__.py

```
@@ -0,0 +1,449 @@
+        uri = (
+            LDAPUrl(uri, scope=ldap.SCOPE_SUBTREE, dn=binddn, cred=bindpw)
+            if not isinstance(uri, LDAPUrl)
+            else uri
+        )
+        self.uri = uri if isinstance(uri, LDAPUrl) else LDAPUrl("ldap:///")
+        self.binddn = binddn
+        self.bindpw = bindpw
+        self.params = params if params is not None else {}
+        self.max = int(max)
+        self._lock = threading.Lock()
+        self._pool: List[Any] = [] # type: ignore
```

Ruff (0.14.8)

Undefined name `List`

(F821)